geode-core/src/main/java/org/apache/geode/internal/security/SecurityServiceFactory.java

```diff
@@ -49,8 +53,21 @@ public static SecurityService create(Properties securityProps, CacheConfig cache
     return create(securityProps, cacheConfig.getSecurityManager(), cacheConfig.getPostProcessor());
   }
 
+  private static SecurityService getTestSecurityService() {
+    String securityServiceClassName = System.getProperty(TEST_SECURITY_SERVICE_SYSTEM_PROPERTY);
+    if (securityServiceClassName == null) {
+      return null;
+    }
+    return CallbackInstantiator.getObjectOfType(securityServiceClassName, SecurityService.class);
+  }
+
   public static SecurityService create(Properties securityProps,
       SecurityManager preferredSecurityManager, PostProcessor preferredPostProcessor) {
+    SecurityService testSecurityService = getTestSecurityService();
+    if (testSecurityService != null) {
+      return testSecurityService;
```

Comment:
If the purpose for this is to just create a specific `SecurityManager` for testing why isn't the normal method, using security properties, an acceptable way to configure a test `SecurityManager`? As written, this test only method completely bypasses all the other security initialization below resulting in a modification of behavior from the normal production behavior.

Reply:
it's create a `SecurityManager`, it's to create a `SecurityService`, the test securityService is a child of the usual security service, only to override one method to throw an exception at intervals.

Reply:
The alternative is that now that we have confirmed that by turning the `UnavailableSecurityManagerException` in the security service into `CacheClosedException` will cause the client to retry other servers, we can get rid of the dunit test and simply use a junit test to make sure the conversion of the exceptions.

Reply:
Probably this other PR is more to the point: https://github.com/apache/geode/pull/7148